<a href="https://colab.research.google.com/github/docNathanM/chemeng316_fluids/blob/main/ChE316_Test_02.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Test 2: ChE 316

In [ ]:
!pip install pint
from pint import UnitRegistry
ur = UnitRegistry()

zsh:1: command not found: pip


ModuleNotFoundError: No module named 'pint'

In [ ]:
from CoolProp.CoolProp import PhaseSI, PropsSI

In [ ]:
import numpy as np

## Problem 1
For the following situations, describe the direction of momentum flux.

## Problem 2

EthylBenzene, C$_8$H$_{10}$, flows at 10 m/s over a boundary surface. A measurement of the boundary layer velocity profile is well estimated by
$$ u(y) = 10 \left[ 1 - e^{-4.605 y / \delta} \right] $$
where $\delta = 5$ mm, and the velocity, $u(y)$, has units of m/s. The viscosity for EthylBenzene in this scenario is found to be $\mu = $ 0.000648 Pa$\cdot$s.

1. Write the definition of wall shear stress, $\tau_{wall}$.
2. Find the velocity gradient, $\partial u / \partial y$, by taking the derivative of the given $u(y)$.
3. Compute the wall shear stress using the results from (a) and (b). Give you final answer in Pascals.


### Given 

In [ ]:
mu = 0.000648 * ur.Pa * ur.sec
U_inf = 10 * ur.meter_per_second
delta_mm = 5 * ur.millimeter


### 2.1 Shear Stress

$$ \tau_{wall} = -\mu \frac{\partial u}{\partial y} $$

### 2.2 Velocity Gradient

$$ 
\frac{\partial u}{\partial y} = \frac{\partial}{\partial y} \left[ 10 - 10 e ^{-4.605 y / \delta} \right]\\
\frac{\partial u}{\partial y} = -10\left[\frac{-4.605}{\delta} e ^{-4.605 y / \delta} \right]\\
\frac{\partial u}{\partial y} = \frac{46.05[\textrm{m}/\textrm{s}]}{\delta[\textrm{mm}] \cdot 1[\textrm{m}]/1000[\textrm{mm}]} e ^{-4.605 y / \delta}
$$



### 2.3 Shear Stress Calculation

$$
\left.\frac{\partial u}{\partial y}\right\vert_{y=0} = \frac{46.05[\textrm{m}/\textrm{s}]}{\delta[\textrm{mm}] \cdot 1[\textrm{m}]/1000[\textrm{mm}]} 
$$

In [ ]:
delta_m = delta_mm.to(ur.meter)
dudy_yZero = 46.05 * ur.meter_per_second / delta_m
tau_wall = - mu * dudy_yZero 

In [ ]:
f"{tau_wall.to(ur.Pa):~P}"

'-5.9680800000000005 Pa'

In [ ]:
f"{dudy_yZero.to(1/ur.second):~P}"

'9210.0 1/s'

**ANSWER** $\tau_{wall} = -5.968$ Pascal

*The negative sign says that the momentum flux is in the negative $y$ direction or into the wall.*

## Problem 3 

In [ ]:
Di = ur.Quantity(2 * ur.inch).to(ur.meter)
mu = 0.000648 * ur.Pa * ur.sec
rho = 869.6 * ur.kilogram / ur.meter**3
[f"{Di.to(ur.inch):~P}",
f"{Di:~P}"]

['2.0 in', '0.0508 m']

In [ ]:
dotVol = 40 * ur.gallon / ur.minute;
dotVol_SI = 0.15 * ur.meter**3 / ur.minute # rounding this off
f"{dotVol_SI:~P}"

'0.15 m³/min'

In [ ]:
A = np.pi * Di**2 / 4
f"{A:~P}"

'0.0020268299163899908 m²'

In [ ]:
Vavg = ur.Quantity(dotVol_SI / A).to(ur.meter_per_second);
f"{Vavg:~P}"

'1.2334532758687407 mps'

In [ ]:
Re = rho * Vavg * Di / mu;
Re.ito(ur.meter/ur.meter)
f"{Re:~P}"

'84087.40310143396'

In [ ]:
k = ur.Quantity(0.00015*ur.feet).to(ur.millimeter) #0.045 * ur.millimeter;
kByD = k.to(ur.meter) / Di;
Cff = -1.8 * np.log10(6.9/Re + (kByD/3.7)**1.11)
ffac = Cff;
f"{ffac:~P}"

'6.7429216496112865'

In [ ]:
L = 50 * ur.meter
delPbyRho_fric = ffac * L / Di * Vavg**2 / 2;
f"{delPbyRho_fric:~P}"

'5048.586657778856 mps²'

In [ ]:
delP_fric = delPbyRho_fric * rho;
delP_fric.ito(ur.Pa)
[f"{delP_fric:~P}", f"{delP_fric.to(ur.psi):~P}"]

['4390250.957604493 Pa', '636.7520669588404 psi']

In [ ]:
k90 = 0.75
kTee = 1.0
k45 = 0.35
kTotal = k90 + kTee + 2*k45
delPbyRho_fittings = kTotal * Vavg**2/2
delP_fittings = delPbyRho_fittings * rho
delP_fittings.ito(ur.Pa)
[f"{delP_fittings:~P}", f"{delP_fittings.to(ur.psi):~P}"]

['1620.6940035109394 Pa', '0.23506179182214243 psi']

In [ ]:
delP_g = rho * ur.Quantity(9.8,ur.meter/ur.second**2) * 30 * ur.meter
print(delP_g.to(ur.Pa))

255662.40000000005 pascal
